### DISCUSSION TITLE
How to sign out an account, which doesn't subscribe to copilot, with Copilot.vim?

### DISCUSSION THREAD

**Discussion by jfcherng** [community/community#126315] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Hi, I am a developer of Sublime Text's copilot plugin [LSP-copilot](https://github.com/TerminalFi/LSP-copilot). The plugin uses [Copilot.vim](https://github.com/github/copilot.vim) (specifically https://github.com/github/copilot.vim/blob/release/dist/agent.js) under the hood.

Today, a user of LSP-copilot reports that he accidentally signed in an account which doesn't subscribe to copilot and then he can't sign out. And I can reproduce that. https://github.com/TerminalFi/LSP-copilot/issues/156

When sign in, `Copilot.vim` creates `C:\Users\jfcherng\AppData\Local\github-copilot\apps.json` on my machine and it reuses that file automatically. But when sign out, it looks like `Copilot.vim` checks whether the account subscribes to copilot. 

```
:: [10:52:02.047] --> LSP-copilot signOut (7): {}
:: [10:52:02.535] <-  LSP-copilot window/logMessage: {'type': 3, 'message': '[auth] Invalid copilot token: missing token: 403 '}
:: [10:52:02.535] <<< LSP-copilot (7) (duration: 488ms): {'status': 'NotAuthorized', 'user': 'XXXXXX'}
```

LSP-copilot simply calls the `signOut` command of `Copilot.vim` but `Copilot.vim` won't do anything if the current account doesn't subscribe to copilot. This results in the user can't change his account.

**Comment by jfcherng** [comment]
I would like to dig more by myself, but https://github.com/github/copilot.vim/blob/release/dist/agent.js doesn't really want me to do that since it doesn't provide readable source codes.

**Reply by jfcherng** [reply]
It looks like `Copilot.vim` just directly calls the `signOut` command as well in https://github.com/github/copilot.vim/blob/c79d711cbf7c6672c6c57d6df7c5ab7b6cac2b7a/autoload/copilot.vim#L630-L638 so I suppose that `Copilot.vim` suffers from the same issue. Can a `Copilot.vim` user confirm this?

**Comment by tpope** [comment]
Can confirm that sign out appears to be broken in Copilot.vim. We'll look into it. As a temporary workaround that `apps.json` can be deleted.

**Comment by tpope** [comment, marked as answer]
Sign out should be fixed in Copilot.vim 1.35.0.

**Reply by jfcherng** [reply]
Thanks!
